### PR TITLE
also check for module wrappers in 'ModulesTool.exist' method

### PR DESCRIPTION
**Pull request by boegel**
fix for #2601 ~~(clearly still WIP, enhanced test will fail in current status)~~

### CODE REVIEW

**Comment by vanzod** [comment]
Tested with `R-3.5.1-foss-2018b.eb`. `Java/1.8` is correctly recognized as installed.

**Comment by boegel** [comment]
Some tests are failing because of problems with the tests themselves, which I'll fix, but one failing test (`test_stale_module_caches`) highlights a problem that should be fixed: we're just checking for an existing `.modulerc` that defines a wrapper, but we're not checking whether the underlying module being wrapped actually exists.

Fixing that is fairly straightforward: check which module file is being wrapped, and then check whether that exists before spitting out a `True` for the existence check of the wrapper. This situation is probably mostly academic in the sense that it's unlikely to happen in practice (unless the module file that the wrapper refers to is removed, of course).

There's a complication though... The wrapped module is specified with the 'short' module name (e.g. `Java/1.8.0_181`) rather than the full module name (e.g. `Core/Java/1.8.0_181`), so we need to "translate" the short name to the full one based on the provided full name of the wrapper module (e.g. `Core/Java/1.8`).

Fixes for both issues coming up in next push (cfr. commits 06621bffc2 + 525d92e03)...

**Comment by boegel** [comment]
@vanzod This should be good to go now...

File: test/framework/modules.py [hunk 1]
```diff
@@ -178,7 +190,7 @@ def test_exists(self):
         self.assertTrue('Core/Java/1.8.0_181' in self.modtool.available())
         self.assertEqual(self.modtool.exist(['Core/Java/1.8.0_181']), [True])
         self.assertEqual(self.modtool.exist(['Core/Java/1.8']), [True])
-        self.assertTrue(self.modtool.module_wrapper_exists('Core/Java/1.8'))
+        self.assertEqual(self.modtool.module_wrapper_exists('Core/Java/1.8'), 'Core/Java/1.8.0_181')
```
Comment (vanzod) [inline test/framework/modules.py]: We have to be careful here if the `Java/1.8` wrapper will point to a different module in the future.

File: test/framework/modules.py [hunk 2]
```diff
@@ -200,7 +213,7 @@ def test_exists(self):
             self.assertTrue('Core/Java/1.8.0_181' in self.modtool.available())
             self.assertEqual(self.modtool.exist(['Core/Java/1.8.0_181']), [True])
             self.assertEqual(self.modtool.exist(['Core/Java/1.8']), [True])
-            self.assertTrue(self.modtool.module_wrapper_exists('Core/Java/1.8'))
+            self.assertEqual(self.modtool.module_wrapper_exists('Core/Java/1.8'), 'Core/Java/1.8.0_181')
```
Comment (vanzod) [inline test/framework/modules.py]: Same as above

File: test/framework/modules.py (hunk 1 above)
Comment (boegel) [reply test/framework/modules.py]: Doesn't matter, this is purely in the unit tests for framework, it has no direct relation to the easyconfigs repository.

A bit higher up in the test I'm making a dummy `Java/1.8.0_181` module file (in a temporary directoty) just for the sake of the test.

So, this won't break if we update the `Java-1.8.eb` easyconfig file in the easyconfigs repo. :)

**Pull request merged**